diff --git label/label.go label/label.go
index ef058b0..affa909 100644
--- label/label.go
+++ label/label.go
@@ -62,10 +62,9 @@ func New(repo, pkg, name string) Label {
 var NoLabel = Label{}
 
 var (
-	labelRepoRegexp = regexp.MustCompile(`^@$|^[A-Za-z.-][A-Za-z0-9_.-]*$`)
-	labelPkgRegexp  = regexp.MustCompile(`^[A-Za-z0-9/._@-]*$`)
-	// This was taken from https://docs.bazel.build/versions/main/build-ref.html#name
-	labelNameRegexp = regexp.MustCompile("^[A-Za-z0-9!%-@^_` \"#$&'()*-+,;<=>?\\[\\]{|}~/.]*$")
+	labelRepoRegexp = regexp.MustCompile(`^@$|^[A-Za-z0-9_.-][A-Za-z0-9_.~-]*$`)
+	labelPkgRegexp  = regexp.MustCompile(`^[\x20-\x39\x3B-\x5B\x5D-\x7E]*$`)
+	labelNameRegexp = labelPkgRegexp
 )
 
 // Parse reads a label from a string.
@@ -91,7 +90,7 @@ func Parse(s string) (Label, error) {
 			s = "//:" + repo
 		}
 		if !labelRepoRegexp.MatchString(repo) {
-			return NoLabel, fmt.Errorf("label parse error: repository has invalid characters: %q", origStr)
+			return NoLabel, fmt.Errorf("label parse error: repository has bad characters: %q and doesn't match regex: %q", origStr, labelRepoRegexp)
 		}
 	}
 
@@ -107,7 +106,7 @@ func Parse(s string) (Label, error) {
 			s = s[endPkg:]
 		}
 		if !labelPkgRegexp.MatchString(pkg) {
-			return NoLabel, fmt.Errorf("label parse error: package has invalid characters: %q", origStr)
+			return NoLabel, fmt.Errorf("label parse error: package has bad characters: %q and doesn't match regex: %q", origStr, labelPkgRegexp)
 		}
 	}
 
